Update vendor to Kubernetes 1.18

dingyin opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://github.com/kubeedge/kubeedge/blob/master/CONTRIBUTING.md
2. Ensure you have added or ran the appropriate tests for your PR

-->

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
/kind feature


**What this PR does / why we need it**:
Update vendor to Kubernetes 1.18

**Which issue(s) this PR fixes**:
<!-- 
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #1933 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note

```

fisherxu commented:
Thanks for the PR, looks like the k8s verion is 1.18.0 and not 1.18.5?

zzxgzgz commented:
> Thanks for the PR, looks like the k8s verion is 1.18.0 and not 1.18.5?

Yes it is, I named my branch incorrectly. Sorry for the confusion.

fisherxu commented:
> Yes it is, I named my branch incorrectly. Sorry for the confusion.

I think we need to bump to the latest version(1.18.6), and please fix the ci issue.

dingyin commented:
> > Yes it is, I named my branch incorrectly. Sorry for the confusion.
> 
> I think we need to bump to the latest version(1.18.6), and please fix the ci issue.

Updated to 1.18.6, and fixed test.

fisherxu commented on cloud/pkg/csidriver/controllerserver.go:
```diff
@@ -20,7 +20,6 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"errors"
-
```
Keep the empty line here :)

dingyin replied:
updated.

fisherxu commented on cloud/pkg/leaderelection/leaderelection.go:
```diff
@@ -1,13 +1,15 @@
 	"syscall"
 
+	"k8s.io/apimachinery/pkg/api/errors"
+
```
delete the empty line

dingyin replied:
updated.

fisherxu commented on edge/pkg/edged/edged.go:
```diff
@@ -698,6 +699,13 @@ func (e *edged) initializeModules() error {
 	}
 
+	// start the CPU manager in the clcm
+	err = e.clcm.StartCPUManager(e.GetActivePods, edgedutil.NewSourcesReady(e.isInitPodReady), e.statusManager, e.runtimeService)
```
If we don't add `StartCPUManager` here, can the ci test pass? If ci can pass, maybe we should add the `feature` in another PR :)

zzxgzgz replied:
If we don't add this function here, the CPU manager won't start, and one of its fields remains nil, which will lead to edgecore crashing. 

That field was not nil in previous k8s versions, but its implementation changed so that the field remains nil until the start function is called.

fisherxu replied:
Ok, and can we move the `StartCPUManager` to Start func instead of `newedged`?

dingyin replied:
fixed.

fisherxu commented on hack/update-vendor-licenses.sh:
```diff
@@ -202,6 +202,13 @@ for PACKAGE in $(go list -m -json all | jq -r .Path | sort -f); do
     echo "${PACKAGE} doesn't exist in ${DEPS_DIR}, skipping" >&2
     continue
   fi
+  if [[ "${PACKAGE}" = "sigs.k8s.io/structured-merge-diff" ]]; then
```
cc @kevin-wangzefeng @daixiang0

fisherxu commented:
@dingyin @zzxgzgz Thanks for the pr, and can you show the memory change after bump to 1.18？

zzxgzgz commented:
> @dingyin @zzxgzgz Thanks for the pr, and can you show the memory change after bump to 1.18？

How do I check it?

fisherxu commented:
> > @dingyin @zzxgzgz Thanks for the pr, and can you show the memory change after bump to 1.18？
> 
> How do I check it?

Run edgecore without this patch and with this patch, then use `top` to record the result :)

dingyin commented:
> @dingyin @zzxgzgz Thanks for the pr, and can you show the memory change after bump to 1.18？

@fisherxu Memory consumption is up from 0.4% to 0.5% in a 16GB box (which is roughly 65MB to 80MB).

fisherxu commented:
/lgtm

kevin-wangzefeng commented:
/approve
